test/quantization/bc/test_backward_compatibility.py

```diff
@@ -142,6 +167,66 @@ def _test_obs(self, obs, input_size, subname=None, generate=False, check_numeric
         if check_numerics:
             self.assertEqual(obs(input_tensor), expected)
 
+    def _test_package(self, fp32_module, input_size, generate=False):
+        """
+        Verifies that files created in the past with torch.package
+        work on today's FX graph mode quantization transforms.
+        """
+        input_file, _state_dict_file, _scripted_module_file, _traced_module_file, \
+            expected_file, package_file, get_attr_targets_file = \
+            get_filenames(self, None)
+
+        package_name = 'test'
+        resource_name_model = 'test.pkl'
+
+        def _do_quant_transforms(
+            m: torch.nn.Module,
+            input_tensor: torch.Tensor,
+        ) -> torch.nn.Module:
+            # do the quantizaton transforms and save result
+            qconfig = torch.quantization.get_default_qconfig('fbgemm')
+            mp = quantize_fx.prepare_fx(m, {'': qconfig})
+            mp(input_tensor)
+            mq = quantize_fx.convert_fx(mp)
+            return mq
+
+        def _get_get_attr_target_strings(m: GraphModule) -> Set[str]:
+            results = set()
+            for node in m.graph.nodes:
+                if node.op == 'get_attr':
+                    results.add(node.target)
+            return results
+
+        if generate and qengine_is_fbgemm():
+            input_tensor = torch.randn(*input_size)
+            torch.save(input_tensor, input_file)
+
+            # save the model with torch.package
+            with torch.package.PackageExporter(package_file) as exp:
+                exp.extern('quantization.bc.test_backward_compatibility')
+                exp.save_pickle(package_name, resource_name_model, fp32_module)
+
+            # do the quantization transforms and save the result
+            mq = _do_quant_transforms(fp32_module, input_tensor)
+            get_attrs = _get_get_attr_target_strings(mq)
+            torch.save(get_attrs, get_attr_targets_file)
+            q_result = mq(input_tensor)
+            torch.save(q_result, expected_file)
+
+        # load input tensor
+        input_tensor = torch.load(input_file)
+        expected_output_tensor = torch.load(expected_file)
+        expected_get_attrs = torch.load(get_attr_targets_file)
+
+        # load model from package and verify output and get_attr targets match
+        imp = torch.package.PackageImporter(package_file)
+        m = imp.load_pickle(package_name, resource_name_model)
+        mq = _do_quant_transforms(m, input_tensor)
+        get_attrs = _get_get_attr_target_strings(mq)
```

Comment:
Can we also get the `prepared.named_buffers()` and compare the observer buffers in the saved model vs the new one?

Reply:
sure